cmake/Extensions.cmake

```diff
@@ -100,6 +100,64 @@ macro(register_extension extension-name extension-display-name extension-guard d
     endif()
 endmacro()
 
+macro(register_c_api_extension extension-name extension-display-name extension-guard description)
+    set(${extension-guard} ${extension-name} PARENT_SCOPE)
+    get_property(extensions GLOBAL PROPERTY EXTENSION-OPTIONS)
+    set_property(GLOBAL APPEND PROPERTY EXTENSION-OPTIONS ${extension-name})
+    target_compile_definitions(${extension-name}
+            PRIVATE "EXTENSION_NAME=${extension-name}" "EXTENSION_VERSION=${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
+    set_target_properties(${extension-name} PROPERTIES
+            ENABLE_EXPORTS True
+            POSITION_INDEPENDENT_CODE ON)
+    if(WIN32)
+        set_target_properties(${extension-name} PROPERTIES
+                RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
+                ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
+                WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
```

Comment:
I would add the export macros in the C API extensions and not export all symbols. CMake's GenerateExportHeader can make it simpler.